### PR TITLE
Use coreutils toolchain for copy_file action

### PR DESCRIPTION
**Pull request by dzbarsky**
Use the coreutils toolchain rather than bash/batch scripts to perform the copy.

### Type of change
- Refactor (a code change that neither fixes a bug or adds a new feature)**

- Relevant documentation has been updated
- Suggested release notes are provided below:
- Breaking change (this change will force users to change their own code or config)
Might require toolchain registration where previously it wasn't needed, if users did their own toolchain registration instead of calling `aspect_bazel_lib_register_toolchains` as the docs recommend

### Test plan
- Covered by existing test cases

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -58,12 +58,12 @@ jobs:
         run: echo "os=macos-latest" >> $GITHUB_OUTPUT
         # Only run on main branch (or PR branches that contain 'macos') to minimize macOS minutes (billed at 10X)
         # https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes
-        if: ${{ github.ref == 'refs/heads/main' || contains(github.head_ref, 'macos') }}
+        #if: ${{ github.ref == 'refs/heads/main' || contains(github.head_ref, 'macos') }}
```
Comment (dzbarsky) [inline .github/workflows/ci.yaml:61]: TODO: remove this before landing

**Comment by gregmagolan** [comment]
Rebased and resolved conflict

**Comment by thesayyn** [comment]
FYI, there was an issue with darwin arm64 binaries for coreutils. https://github.com/aspect-build/bazel-lib/issues/519

**Comment by gregmagolan** [comment]
> FYI, there was an issue with darwin arm64 binaries for coreutils. #519

So this PR will break copy_file on M1s until that is resolved?

**Comment by thesayyn** [comment]
> So this PR will break copy_file on M1s until that is resolved?

yes, if you don't have rosetta turned on

**Comment by gregmagolan** [comment]
Roger. We'll have to wait until that is resolved then before we can land this one.

**Comment by dzbarsky** [comment]
Let's see how they feel about https://github.com/uutils/coreutils/pull/5523

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (thesayyn) [reply .github/workflows/ci.yaml:61]: revert?

**Review by thesayyn — APPROVED** [review]

**Review by kormide — APPROVED** [review]

**Pull request merged**